api/src/main/java/org/pmiops/workbench/api/ProfileController.java

```diff
@@ -264,15 +264,17 @@ private String createFirecloudBillingProject(User user) {
   private User initializeUserIfNeeded() {
     UserAuthentication userAuthentication = userAuthenticationProvider.get();
     User user = userAuthentication.getUser();
+    Timestamp now = new Timestamp(clock.instant().toEpochMilli());
     if (userAuthentication.getUserType() == UserType.SERVICE_ACCOUNT) {
       // Service accounts don't need further initialization.
       return user;
     }
 
-    Boolean twoFactorEnabled = Optional.ofNullable(user.getTwoFactorEnabled()).orElse(false);
-    if (!twoFactorEnabled) {
-      user.setTwoFactorEnabled(directoryService.getUser(user.getEmail()).getIsEnrolledIn2Sv());
-      user = saveUserWithConflictHandling(user);
+    if (user.getTwoFactorAuthCompletionTime() == null) {
```

Comment:
nit: we can make this slightly more readable by adding a `completedTwoFactorAuth()` function to the User class which does this check

Reply:
i'm going to argue that that makes this less readable because i have to go click around more, and also adds more boilerplate code

Reply:
That's fine. But in regards to clicking around, the goal for me is that I shouldn't have to. I don't really care how its implemented, I just want to know if its enabled or not which I should be able to glean from the function name. I had to read through about half the PR before I understood the purpose of the 2fa completion time, and then was able to come back here to reread it and understand whats going on.